application-analytics-default/src/main/java/com/xwiki/analytics/internal/aggregators/AggregatorSaver.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.analytics.internal.aggregators;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.SpaceReference;
+import org.xwiki.rendering.syntax.Syntax;
+
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+/**
+ * Handles the storage of aggregated data.
+ *
+ * @version $Id$
+ * @since 1.13
+ */
+@Component(roles = AggregatorSaver.class)
+@Singleton
+public class AggregatorSaver
+{
+    private static final List<String> SPACE = List.of("Analytics", "Code", "AggregateData");
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private Logger logger;
+
+    /**
+     * Saves the aggregated data on a xwiki page.
+     *
+     * @param data data that you want to store
+     * @param pageName name of the page where you want your data
+     */
+    public void saveData(ArrayNode data, String pageName)
+    {
+        XWikiContext context = contextProvider.get();
+        XWiki xwiki = context.getWiki();
+        SpaceReference spaceReference = new SpaceReference(context.getWikiId(), SPACE);
+        DocumentReference documentReference = new DocumentReference(pageName, spaceReference);
+        try {
+            // I don't really know witch user should be used when saving the data
```

Comment:
I am not exactly sure which user should be displayed when saving the history. By default, the user who executes the Scheduler job is the one who created the job page